packages/query/src/graphql/plugins/GetMetadataPlugin.ts

```diff
@@ -25,27 +25,65 @@ const metaCache = {
   queryNodeVersion: packageVersion,
 } as Metadata;
 
-export const GetMetadataPlugin = makeExtendSchemaPlugin((build) => {
-  setAsyncInterval(async () => {
-    let health;
-    let meta;
-    try {
-      meta = await fetch(new URL(`meta`, indexerUrl));
-      const result = await meta.json();
-      Object.assign(metaCache, result);
-    } catch (e) {
-      metaCache.indexerHealthy = false;
-      console.warn(`Failed to fetch indexer meta, `, e.message);
-    }
-
-    try {
-      health = await fetch(new URL(`health`, indexerUrl));
-      metaCache.indexerHealthy = !!health.ok;
-    } catch (e) {
-      metaCache.indexerHealthy = false;
-      console.warn(`Failed to fetch indexer health, `, e.message);
-    }
-  }, 10000);
+async function fetchFromApi(): Promise<void> {
+  let health: Response;
+  let meta: Response;
+
+  try {
+    meta = await fetch(new URL(`meta`, indexerUrl));
+    const result = await meta.json();
+    Object.assign(metaCache, result);
+  } catch (e) {
+    metaCache.indexerHealthy = false;
+    console.warn(`Failed to fetch indexer meta, `, e.message);
+  }
+
+  try {
+    health = await fetch(new URL(`health`, indexerUrl));
+    metaCache.indexerHealthy = !!health.ok;
+  } catch (e) {
+    metaCache.indexerHealthy = false;
+    console.warn(`Failed to fetch indexer health, `, e.message);
+  }
+}
+
+function fetchFromTable(rows): Metadata {
```

Comment:
Is this ok @ianhe8x, It will prevent wrong field from being added and wrong value types.

Reply:
not it doesn't. Typescript is not strong type language, it doesn't check field or value type for you.